Fix testRunner.displayAndTrackRepaints() for site isolation and remove `WKBundlePageSetTracksRepaints`/`WKBundlePageResetTrackedRepaints`

charliewolfe opened the pull request:
#### bd92e5e29350527a9e198dc521ae782ed015e880
<pre>
Fix testRunner.displayAndTrackRepaints() for site isolation and remove `WKBundlePageSetTracksRepaints`/`WKBundlePageResetTrackedRepaints`
<a href="https://bugs.webkit.org/show_bug.cgi?id=282415">https://bugs.webkit.org/show_bug.cgi?id=282415</a>
<a href="https://rdar.apple.com/139033191">rdar://139033191</a>

Reviewed by Alex Christensen.

To behave as expected when using --site-isolation, testRunner.displayAndTrackRepaints() needs to force a
repaint and enable repaint tracking on frames hosted in other processes.

Also remove all other uses of WKBundlePageSetTracksRepaints and WKBundlePageResetTrackedRepaints,
replacing them with the site isolation compatible WKPage C API.

* LayoutTests/compositing/clip-change.html:
* LayoutTests/compositing/geometry/foreground-offset-change.html:
* LayoutTests/compositing/geometry/limit-layer-bounds-transformed-overflow.html:
* LayoutTests/compositing/iframes/become-composited-nested-iframes.html:
* LayoutTests/compositing/iframes/become-overlapped-iframe.html:
* LayoutTests/compositing/iframes/connect-compositing-iframe-delayed.html:
* LayoutTests/compositing/iframes/connect-compositing-iframe.html:
* LayoutTests/compositing/iframes/connect-compositing-iframe2.html:
* LayoutTests/compositing/iframes/connect-compositing-iframe3.html:
* LayoutTests/compositing/iframes/enter-compositing-iframe.html:
* LayoutTests/compositing/iframes/iframe-content-flipping.html:
* LayoutTests/compositing/iframes/iframe-resize.html:
* LayoutTests/compositing/iframes/iframe-scrolling-change.html:
* LayoutTests/compositing/iframes/iframe-size-to-zero.html:
* LayoutTests/compositing/iframes/iframe-src-change.html:
* LayoutTests/compositing/iframes/invisible-nested-iframe-hide.html:
* LayoutTests/compositing/iframes/invisible-nested-iframe-show.html:
* LayoutTests/compositing/iframes/leave-compositing-iframe.html:
* LayoutTests/compositing/iframes/overlapped-iframe-iframe.html:
* LayoutTests/compositing/iframes/overlapped-iframe.html:
* LayoutTests/compositing/iframes/overlapped-nested-iframes.html:
* LayoutTests/compositing/iframes/repaint-after-losing-scrollbars.html:
* LayoutTests/compositing/iframes/resources/fixed-position-subframe.html:
* LayoutTests/compositing/iframes/resources/fixed-position-transformed-subframe.html:
* LayoutTests/compositing/iframes/resources/scrollgrandchild-inner.html:
* LayoutTests/compositing/iframes/scrolling-iframe.html:
* LayoutTests/compositing/layer-creation/change-to-overlap.html:
* LayoutTests/compositing/overflow/content-gains-scrollbars.html:
* LayoutTests/compositing/overflow/content-loses-scrollbars.html:
* LayoutTests/compositing/overflow/do-not-paint-outline-into-composited-scrolling-contents.html:
* LayoutTests/compositing/overflow/overflow-scrollbar-layers.html:
* LayoutTests/compositing/overflow/paint-neg-z-order-descendants-into-scrolling-contents-layer.html:
* LayoutTests/compositing/overflow/repaint-after-losing-scrollbars.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-add-text-expected.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-add-text.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-change-expected.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-change.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-load-expected.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-load.html:
* LayoutTests/compositing/repaint/iframes/composited-iframe-with-fixed-background-doc-repaint.html:
* LayoutTests/compositing/repaint/iframes/compositing-iframe-scroll-repaint.html:
* LayoutTests/compositing/repaint/iframes/compositing-iframe-with-fixed-background-doc-repaint.html:
* LayoutTests/compositing/repaint/newly-composited-on-scroll.html:
* LayoutTests/compositing/repaint/newly-composited-repaint-rect.html:
* LayoutTests/compositing/repaint/page-scale-repaint.html:
* LayoutTests/compositing/repaint/requires-backing-repaint.html:
* LayoutTests/compositing/repaint/transform-style-change.html:
* LayoutTests/compositing/rtl/rtl-overflow-invalidation.html:
* LayoutTests/compositing/scroll-painted-composited-content.html:
* LayoutTests/compositing/tiling/huge-layer-resize.html:
* LayoutTests/compositing/video-page-visibility.html:
* LayoutTests/compositing/video/video-with-invalid-source.html:
* LayoutTests/compositing/webgl/webgl-repaint.html:
* LayoutTests/css3/filters/multiple-filters-invalidation.html:
* LayoutTests/css3/flexbox/repaint-column-reverse.html:
* LayoutTests/css3/flexbox/repaint-during-resize-no-flex.html:
* LayoutTests/css3/flexbox/repaint-rtl-column.html:
* LayoutTests/css3/flexbox/repaint.html:
* LayoutTests/displaylists/resources/dump-target-display-list.js:
(async doTest):
(doTest): Deleted.
* LayoutTests/displaylists/resources/dump-target-replay-list.js:
(async doTest):
(doTest): Deleted.
* LayoutTests/fast/animation/request-animation-frame-cancel.html:
* LayoutTests/fast/animation/request-animation-frame-detach-element.html:
* LayoutTests/fast/animation/request-animation-frame-detach-element2.html:
* LayoutTests/fast/animation/request-animation-frame-iframe.html:
* LayoutTests/fast/animation/request-animation-frame-iframe2.html:
* LayoutTests/fast/animation/request-animation-frame-time-unit.html:
* LayoutTests/fast/animation/request-animation-frame-timestamps-advance.html:
* LayoutTests/fast/animation/request-animation-frame-within-callback.html:
* LayoutTests/fast/animation/request-animation-frame.html:
* LayoutTests/fast/animation/script-tests/request-animation-frame-subframe.html:
* LayoutTests/fast/backgrounds/obscured-background-child-style-change-expected.html:
* LayoutTests/fast/backgrounds/obscured-background-child-style-change.html:
* LayoutTests/fast/backgrounds/solid-color-context-restore.html:
* LayoutTests/fast/canvas/canvas-as-image-incremental-repaint.html:
* LayoutTests/fast/canvas/canvas-imageSmoothingEnabled-repaint.html:
* LayoutTests/fast/canvas/canvas-incremental-repaint-2.html:
* LayoutTests/fast/canvas/canvas-incremental-repaint.html:
* LayoutTests/fast/canvas/canvas-resize-after-paint-without-layout.html:
* LayoutTests/fast/canvas/canvas-resize-after-paint.html:
* LayoutTests/fast/canvas/resources/repaint.js:
(async runRepaintTest):
(runRepaintTest): Deleted.
* LayoutTests/fast/canvas/webgl/canvas-test.html:
* LayoutTests/fast/canvas/webgl/webgl-layer-update.html:
* LayoutTests/fast/css/pseudostyle-anonymous-text.html:
* LayoutTests/fast/css/relative-positioned-block-crash.html:
* LayoutTests/fast/dom/Window/web-timing-minimal-performance-now.html:
* LayoutTests/fast/dynamic/containing-block-change.html:
* LayoutTests/fast/dynamic/layer-no-longer-paginated.html:
* LayoutTests/fast/forms/hidden-listbox.html:
* LayoutTests/fast/forms/listbox-clip.html:
* LayoutTests/fast/frames/inline-object-inside-frameset.html:
* LayoutTests/fast/frames/transparent-scrollbar.html:
* LayoutTests/fast/images/animated-background-image-crash.html:
* LayoutTests/fast/layers/add-layer-with-nested-stacking.html:
* LayoutTests/fast/layers/inline-dirty-z-order-lists.html:
* LayoutTests/fast/layers/layer-content-visibility-change.html:
* LayoutTests/fast/layers/nested-layers-1.html:
* LayoutTests/fast/layers/nested-layers-2.html:
* LayoutTests/fast/layers/nested-layers-3.html:
* LayoutTests/fast/layers/nested-layers-4.html:
* LayoutTests/fast/layers/remove-layer-with-nested-stacking.html:
* LayoutTests/fast/overflow/hidden-scrollbar-resize.html:
* LayoutTests/fast/reflections/inline-crash.html:
* LayoutTests/fast/repaint/animation-after-layer-scroll.html:
* LayoutTests/fast/repaint/background-shorthand-with-gradient-and-height-changes.html:
* LayoutTests/fast/repaint/canvas-object-fit.html:
* LayoutTests/fast/repaint/fixed-child-move-after-scroll.html:
* LayoutTests/fast/repaint/fixed-child-of-fixed-move-after-scroll.html:
* LayoutTests/fast/repaint/fixed-child-of-transformed-move-after-scroll.html:
* LayoutTests/fast/repaint/fixed-move-after-scroll.html:
* LayoutTests/fast/repaint/no-animation-outside-viewport-subframe.html:
* LayoutTests/fast/repaint/no-animation-outside-viewport.html:
* LayoutTests/fast/repaint/obscured-background-no-repaint.html:
* LayoutTests/fast/repaint/overflow-hidden-in-overflow-hidden-scrolled.html:
* LayoutTests/fast/repaint/renderer-destruction-by-invalidateSelection-crash.html:
* LayoutTests/fast/repaint/repaint-during-scroll-with-zoom.html:
* LayoutTests/fast/repaint/resources/fixed-move-after-keyboard-scroll-iframe.html:
* LayoutTests/fast/repaint/resources/iframe-scroll-repaint-iframe.html:
* LayoutTests/fast/repaint/resources/line-flow-with-floats.js:
(async loaded):
(async beginTest):
(loaded): Deleted.
(beginTest): Deleted.
* LayoutTests/fast/repaint/table-overflow-hidden-in-overflow-hidden-scrolled.html:
* LayoutTests/fast/repaint/translate-animation-repaint.html:
* LayoutTests/fast/replaced/width100percent-textarea.html:
* LayoutTests/fast/table/border-collapsing/cached-69296.html:
* LayoutTests/fast/table/border-collapsing/cached-cell-append.html:
* LayoutTests/fast/table/border-collapsing/cached-cell-remove.html:
* LayoutTests/fast/table/border-collapsing/cached-change-cell-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-cell-border-width.html:
* LayoutTests/fast/table/border-collapsing/cached-change-cell-sl-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-col-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-col-border-width.html:
* LayoutTests/fast/table/border-collapsing/cached-change-colgroup-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-colgroup-border-width.html:
* LayoutTests/fast/table/border-collapsing/cached-change-row-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-row-border-width.html:
* LayoutTests/fast/table/border-collapsing/cached-change-table-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-table-border-width.html:
* LayoutTests/fast/table/border-collapsing/cached-change-tbody-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-tbody-border-width.html:
* LayoutTests/fast/table/empty-section-crash.html:
* LayoutTests/fast/text/font-initial.html:
* LayoutTests/fast/text/international/harfbuzz-runs-with-no-glyph.html:
* LayoutTests/fast/text/text-shadow-extreme-value.html:
* LayoutTests/http/tests/misc/slow-loading-image-in-pattern.html:
* LayoutTests/http/tests/site-isolation/compositing/iframes/iframe-content-flipping.html:
* LayoutTests/http/tests/site-isolation/compositing/iframes/iframe-scrolling-change.html:
* LayoutTests/scrollbars/scrollbar-percent-padding-crash.html:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetTracksRepaintsForTesting):
(WKPageDisplayAndTrackRepaintsForTesting):
* Source/WebKit/UIProcess/API/C/WKPagePrivate.h:
* Source/WebKit/UIProcess/WebPageProxyTesting.cpp:
(WebKit::WebPageProxyTesting::setTracksRepaints):
(WebKit::WebPageProxyTesting::displayAndTrackRepaints):
* Source/WebKit/UIProcess/WebPageProxyTesting.h:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageSetTracksRepaints): Deleted.
(WKBundlePageResetTrackedRepaints): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePagePrivate.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::setTracksRepaints): Deleted.
(WebKit::WebPage::resetTrackedRepaints): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPageTesting.cpp:
(WebKit::WebPageTesting::setTracksRepaints):
(WebKit::WebPageTesting::displayAndTrackRepaints):
* Source/WebKit/WebProcess/WebPage/WebPageTesting.h:
* Source/WebKit/WebProcess/WebPage/WebPageTesting.messages.in:
* Source/WebKit/mac/WebKit2.order:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::beginTesting):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::InjectedBundlePage::prepare): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.h:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::displayAndTrackRepaints):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::resetStateToConsistentValues):
(WTR::TestController::didReceiveAsyncMessageFromInjectedBundle):
(WTR::TestController::setTracksRepaints):
* Tools/WebKitTestRunner/TestController.h:

Canonical link: <a href="https://commits.webkit.org/286103@main">https://commits.webkit.org/286103@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/65f51994b5858ce609e0c93c5186085659bead7d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74527 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53956 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/27338 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78975 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25765 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/76644 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/63089 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1741 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58603 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16867 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77594 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48722 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/64082 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38995 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45788 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21579 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/24098 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/67134 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21924 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/80435 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1844 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/1089 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66843 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1992 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/64100 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66128 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10066 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/8226 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11545 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1809 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/4596 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1837 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/2794 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1844 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Tools/WebKitTestRunner/TestController.cpp:
```diff
@@ -4292,6 +4297,13 @@ void TestController::setServiceWorkerFetchTimeoutForTesting(double seconds)
     WKWebsiteDataStoreSetServiceWorkerFetchTimeoutForTesting(websiteDataStore(), seconds);
 }
 
+void TestController::setTracksRepaints(bool trackRepaints)
```
This is only called with false.  We could call it resetTracksRepaints() to prevent someone in the future from using this with true when the completion handler version might be slightly more efficient.  But this is fine.

charliewolfe committed:
Fix testRunner.displayAndTrackRepaints() for site isolation and remove `WKBundlePageSetTracksRepaints`/`WKBundlePageResetTrackedRepaints`
https://bugs.webkit.org/show_bug.cgi?id=282415
rdar://139033191

Reviewed by Alex Christensen.

To behave as expected when using --site-isolation, testRunner.displayAndTrackRepaints() needs to force a
repaint and enable repaint tracking on frames hosted in other processes.

Also remove all other uses of WKBundlePageSetTracksRepaints and WKBundlePageResetTrackedRepaints,
replacing them with the site isolation compatible WKPage C API.

* LayoutTests/compositing/clip-change.html:
* LayoutTests/compositing/geometry/foreground-offset-change.html:
* LayoutTests/compositing/geometry/limit-layer-bounds-transformed-overflow.html:
* LayoutTests/compositing/iframes/become-composited-nested-iframes.html:
* LayoutTests/compositing/iframes/become-overlapped-iframe.html:
* LayoutTests/compositing/iframes/connect-compositing-iframe-delayed.html:
* LayoutTests/compositing/iframes/connect-compositing-iframe.html:
* LayoutTests/compositing/iframes/connect-compositing-iframe2.html:
* LayoutTests/compositing/iframes/connect-compositing-iframe3.html:
* LayoutTests/compositing/iframes/enter-compositing-iframe.html:
* LayoutTests/compositing/iframes/iframe-content-flipping.html:
* LayoutTests/compositing/iframes/iframe-resize.html:
* LayoutTests/compositing/iframes/iframe-scrolling-change.html:
* LayoutTests/compositing/iframes/iframe-size-to-zero.html:
* LayoutTests/compositing/iframes/iframe-src-change.html:
* LayoutTests/compositing/iframes/invisible-nested-iframe-hide.html:
* LayoutTests/compositing/iframes/invisible-nested-iframe-show.html:
* LayoutTests/compositing/iframes/leave-compositing-iframe.html:
* LayoutTests/compositing/iframes/overlapped-iframe-iframe.html:
* LayoutTests/compositing/iframes/overlapped-iframe.html:
* LayoutTests/compositing/iframes/overlapped-nested-iframes.html:
* LayoutTests/compositing/iframes/repaint-after-losing-scrollbars.html:
* LayoutTests/compositing/iframes/resources/fixed-position-subframe.html:
* LayoutTests/compositing/iframes/resources/fixed-position-transformed-subframe.html:
* LayoutTests/compositing/iframes/resources/scrollgrandchild-inner.html:
* LayoutTests/compositing/iframes/scrolling-iframe.html:
* LayoutTests/compositing/layer-creation/change-to-overlap.html:
* LayoutTests/compositing/overflow/content-gains-scrollbars.html:
* LayoutTests/compositing/overflow/content-loses-scrollbars.html:
* LayoutTests/compositing/overflow/do-not-paint-outline-into-composited-scrolling-contents.html:
* LayoutTests/compositing/overflow/overflow-scrollbar-layers.html:
* LayoutTests/compositing/overflow/paint-neg-z-order-descendants-into-scrolling-contents-layer.html:
* LayoutTests/compositing/overflow/repaint-after-losing-scrollbars.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-add-text-expected.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-add-text.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-change-expected.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-change.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-load-expected.html:
* LayoutTests/compositing/patterns/direct-pattern-compositing-load.html:
* LayoutTests/compositing/repaint/iframes/composited-iframe-with-fixed-background-doc-repaint.html:
* LayoutTests/compositing/repaint/iframes/compositing-iframe-scroll-repaint.html:
* LayoutTests/compositing/repaint/iframes/compositing-iframe-with-fixed-background-doc-repaint.html:
* LayoutTests/compositing/repaint/newly-composited-on-scroll.html:
* LayoutTests/compositing/repaint/newly-composited-repaint-rect.html:
* LayoutTests/compositing/repaint/page-scale-repaint.html:
* LayoutTests/compositing/repaint/requires-backing-repaint.html:
* LayoutTests/compositing/repaint/transform-style-change.html:
* LayoutTests/compositing/rtl/rtl-overflow-invalidation.html:
* LayoutTests/compositing/scroll-painted-composited-content.html:
* LayoutTests/compositing/tiling/huge-layer-resize.html:
* LayoutTests/compositing/video-page-visibility.html:
* LayoutTests/compositing/video/video-with-invalid-source.html:
* LayoutTests/compositing/webgl/webgl-repaint.html:
* LayoutTests/css3/filters/multiple-filters-invalidation.html:
* LayoutTests/css3/flexbox/repaint-column-reverse.html:
* LayoutTests/css3/flexbox/repaint-during-resize-no-flex.html:
* LayoutTests/css3/flexbox/repaint-rtl-column.html:
* LayoutTests/css3/flexbox/repaint.html:
* LayoutTests/displaylists/resources/dump-target-display-list.js:
(async doTest):
(doTest): Deleted.
* LayoutTests/displaylists/resources/dump-target-replay-list.js:
(async doTest):
(doTest): Deleted.
* LayoutTests/fast/animation/request-animation-frame-cancel.html:
* LayoutTests/fast/animation/request-animation-frame-detach-element.html:
* LayoutTests/fast/animation/request-animation-frame-detach-element2.html:
* LayoutTests/fast/animation/request-animation-frame-iframe.html:
* LayoutTests/fast/animation/request-animation-frame-iframe2.html:
* LayoutTests/fast/animation/request-animation-frame-time-unit.html:
* LayoutTests/fast/animation/request-animation-frame-timestamps-advance.html:
* LayoutTests/fast/animation/request-animation-frame-within-callback.html:
* LayoutTests/fast/animation/request-animation-frame.html:
* LayoutTests/fast/animation/script-tests/request-animation-frame-subframe.html:
* LayoutTests/fast/backgrounds/obscured-background-child-style-change-expected.html:
* LayoutTests/fast/backgrounds/obscured-background-child-style-change.html:
* LayoutTests/fast/backgrounds/solid-color-context-restore.html:
* LayoutTests/fast/canvas/canvas-as-image-incremental-repaint.html:
* LayoutTests/fast/canvas/canvas-imageSmoothingEnabled-repaint.html:
* LayoutTests/fast/canvas/canvas-incremental-repaint-2.html:
* LayoutTests/fast/canvas/canvas-incremental-repaint.html:
* LayoutTests/fast/canvas/canvas-resize-after-paint-without-layout.html:
* LayoutTests/fast/canvas/canvas-resize-after-paint.html:
* LayoutTests/fast/canvas/resources/repaint.js:
(async runRepaintTest):
(runRepaintTest): Deleted.
* LayoutTests/fast/canvas/webgl/canvas-test.html:
* LayoutTests/fast/canvas/webgl/webgl-layer-update.html:
* LayoutTests/fast/css/pseudostyle-anonymous-text.html:
* LayoutTests/fast/css/relative-positioned-block-crash.html:
* LayoutTests/fast/dom/Window/web-timing-minimal-performance-now.html:
* LayoutTests/fast/dynamic/containing-block-change.html:
* LayoutTests/fast/dynamic/layer-no-longer-paginated.html:
* LayoutTests/fast/forms/hidden-listbox.html:
* LayoutTests/fast/forms/listbox-clip.html:
* LayoutTests/fast/frames/inline-object-inside-frameset.html:
* LayoutTests/fast/frames/transparent-scrollbar.html:
* LayoutTests/fast/images/animated-background-image-crash.html:
* LayoutTests/fast/layers/add-layer-with-nested-stacking.html:
* LayoutTests/fast/layers/inline-dirty-z-order-lists.html:
* LayoutTests/fast/layers/layer-content-visibility-change.html:
* LayoutTests/fast/layers/nested-layers-1.html:
* LayoutTests/fast/layers/nested-layers-2.html:
* LayoutTests/fast/layers/nested-layers-3.html:
* LayoutTests/fast/layers/nested-layers-4.html:
* LayoutTests/fast/layers/remove-layer-with-nested-stacking.html:
* LayoutTests/fast/overflow/hidden-scrollbar-resize.html:
* LayoutTests/fast/reflections/inline-crash.html:
* LayoutTests/fast/repaint/animation-after-layer-scroll.html:
* LayoutTests/fast/repaint/background-shorthand-with-gradient-and-height-changes.html:
* LayoutTests/fast/repaint/canvas-object-fit.html:
* LayoutTests/fast/repaint/fixed-child-move-after-scroll.html:
* LayoutTests/fast/repaint/fixed-child-of-fixed-move-after-scroll.html:
* LayoutTests/fast/repaint/fixed-child-of-transformed-move-after-scroll.html:
* LayoutTests/fast/repaint/fixed-move-after-scroll.html:
* LayoutTests/fast/repaint/no-animation-outside-viewport-subframe.html:
* LayoutTests/fast/repaint/no-animation-outside-viewport.html:
* LayoutTests/fast/repaint/obscured-background-no-repaint.html:
* LayoutTests/fast/repaint/overflow-hidden-in-overflow-hidden-scrolled.html:
* LayoutTests/fast/repaint/renderer-destruction-by-invalidateSelection-crash.html:
* LayoutTests/fast/repaint/repaint-during-scroll-with-zoom.html:
* LayoutTests/fast/repaint/resources/fixed-move-after-keyboard-scroll-iframe.html:
* LayoutTests/fast/repaint/resources/iframe-scroll-repaint-iframe.html:
* LayoutTests/fast/repaint/resources/line-flow-with-floats.js:
(async loaded):
(async beginTest):
(loaded): Deleted.
(beginTest): Deleted.
* LayoutTests/fast/repaint/table-overflow-hidden-in-overflow-hidden-scrolled.html:
* LayoutTests/fast/repaint/translate-animation-repaint.html:
* LayoutTests/fast/replaced/width100percent-textarea.html:
* LayoutTests/fast/table/border-collapsing/cached-69296.html:
* LayoutTests/fast/table/border-collapsing/cached-cell-append.html:
* LayoutTests/fast/table/border-collapsing/cached-cell-remove.html:
* LayoutTests/fast/table/border-collapsing/cached-change-cell-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-cell-border-width.html:
* LayoutTests/fast/table/border-collapsing/cached-change-cell-sl-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-col-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-col-border-width.html:
* LayoutTests/fast/table/border-collapsing/cached-change-colgroup-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-colgroup-border-width.html:
* LayoutTests/fast/table/border-collapsing/cached-change-row-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-row-border-width.html:
* LayoutTests/fast/table/border-collapsing/cached-change-table-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-table-border-width.html:
* LayoutTests/fast/table/border-collapsing/cached-change-tbody-border-color.html:
* LayoutTests/fast/table/border-collapsing/cached-change-tbody-border-width.html:
* LayoutTests/fast/table/empty-section-crash.html:
* LayoutTests/fast/text/font-initial.html:
* LayoutTests/fast/text/international/harfbuzz-runs-with-no-glyph.html:
* LayoutTests/fast/text/text-shadow-extreme-value.html:
* LayoutTests/http/tests/misc/slow-loading-image-in-pattern.html:
* LayoutTests/http/tests/site-isolation/compositing/iframes/iframe-content-flipping.html:
* LayoutTests/http/tests/site-isolation/compositing/iframes/iframe-scrolling-change.html:
* LayoutTests/scrollbars/scrollbar-percent-padding-crash.html:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetTracksRepaintsForTesting):
(WKPageDisplayAndTrackRepaintsForTesting):
* Source/WebKit/UIProcess/API/C/WKPagePrivate.h:
* Source/WebKit/UIProcess/WebPageProxyTesting.cpp:
(WebKit::WebPageProxyTesting::setTracksRepaints):
(WebKit::WebPageProxyTesting::displayAndTrackRepaints):
* Source/WebKit/UIProcess/WebPageProxyTesting.h:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageSetTracksRepaints): Deleted.
(WKBundlePageResetTrackedRepaints): Deleted.
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePagePrivate.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::setTracksRepaints): Deleted.
(WebKit::WebPage::resetTrackedRepaints): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPageTesting.cpp:
(WebKit::WebPageTesting::setTracksRepaints):
(WebKit::WebPageTesting::displayAndTrackRepaints):
* Source/WebKit/WebProcess/WebPage/WebPageTesting.h:
* Source/WebKit/WebProcess/WebPage/WebPageTesting.messages.in:
* Source/WebKit/mac/WebKit2.order:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::beginTesting):
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::InjectedBundlePage::prepare): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.h:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::displayAndTrackRepaints):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::resetStateToConsistentValues):
(WTR::TestController::didReceiveAsyncMessageFromInjectedBundle):
(WTR::TestController::setTracksRepaints):
* Tools/WebKitTestRunner/TestController.h:

Canonical link: https://commits.webkit.org/286103@main
